packages/core/src/mixins/renderToLayer.js

```diff
@@ -33,80 +10,57 @@ import '../styles/RenderToLayer.scss';
 const COMPONENT_NAME = prefixClass('base-layer');
 const LAYER_ID_PREFIX = 'layer';
 
+export function createLayer() {
+    const layer = document.createElement('div');
+    layer.className = COMPONENT_NAME;
+    layer.id = randId({ prefix: LAYER_ID_PREFIX });
+
+    return layer;
+}
+
+/**
+ * renderToLayer() HOC mixin
+ * =========================
+ * Render a component to an node outside of React root,
+ * using `React.createPortal()`.
+ *
+ * @param {React.ComponentType<any>} WrappedComponent
+ *
+ * @example
+ * const ExternalComponent = renderToLayer(Component);
+ */
 function renderToLayer(WrappedComponent) {
     const componentName = getComponentName(WrappedComponent);
 
     class RenderToLayer extends Component {
         static displayName = `renderToLayer(${componentName})`;
 
-        constructor(...args) {
-            super(...args);
+        state = {
+            inDOM: false,
+        };
 
-            this.baseLayer = null;
-            this.componentRef = null;
+        constructor(props) {
+            super(props);
+            this.baseLayer = createLayer();
         }
 
         componentDidMount() {
-            this.createBaseLayer();
-            this.renderComponentToLayer();
-        }
-
-        componentWillReceiveProps(nextProps) {
-            this.renderComponentToLayer({ withProps: nextProps });
+            document.body.appendChild(this.baseLayer);
+            this.setState({ inDOM: true });
```

Comment:
Comments related to the problems encountered in the anchor could be added here

Reply:
The state approach is recommended by React document
https://reactjs.org/docs/portals.html